scripts/mbedtls_framework/code_wrapper/psa_wrapper.py

```diff
@@ -0,0 +1,257 @@
+        'mbedtls_psa_register_se_key': 'defined(MBEDTLS_PSA_CRYPTO_SE_C)',
+        'mbedtls_psa_inject_entropy': 'defined(MBEDTLS_PSA_INJECT_ENTROPY)',
+        'mbedtls_psa_external_get_random': 'defined(MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG)',
+        'mbedtls_psa_platform_get_builtin_key': 'defined(MBEDTLS_PSA_CRYPTO_BUILTIN_KEYS)',
+        'psa_crypto_driver_pake_get_cipher_suite' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_password' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_password_len' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_peer' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_peer_len' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_user' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_crypto_driver_pake_get_user_len' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_abort' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_get_implicit_key' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_input' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_output' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_password_key' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_peer' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_role' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_user' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_setup' : 'defined(PSA_WANT_ALG_SOME_PAKE)'
+    }
+}
+
+class PSAWrapper(c_wrapper_generator.Base):
+    """Generate a C source file containing wrapper functions for PSA Crypto API calls."""
+
+    _WRAPPER_NAME_PREFIX = 'mbedtls_test_wrap_'
+    _WRAPPER_NAME_SUFFIX = ''
+
+    _PSA_WRAPPER_INCLUDES = ['<psa/crypto.h>']
+
+    def __init__(self,
+                 output_h_f: str,
+                 output_c_f: str,
+                 in_headers:  Collection[str] = DEFAULTS["input_headers"],
+                 config: Dict[str, Any]= DEFAULTS) -> None:
+
+        super().__init__()
+        self._FUNCTION_GUARDS = super()._FUNCTION_GUARDS.copy()
+        self.in_headers = in_headers
+        self.out_c_f = output_c_f
+        self.out_h_f = output_h_f
+
+        self.mbedtls_root = build_tree.guess_mbedtls_root()
+        self.read_config(config)
+
+        if in_headers:
+            self.read_headers(in_headers)
+
+    def read_config(self, cfg: Dict[str, Any])-> None:
+        """Configure instance's parameters based on a module specific default config """
+
+        self._CPP_GUARDS = PSAWrapper.parse_def_guards(cfg["define_guards"])
+        self._SKIP_FUNCTIONS = cfg["skip_list"]
+        self._FUNCTION_GUARDS.update(cfg["function_guards"]) # type: ignore[arg-type]
+        self._NOT_IMPLEMENTED = cfg["not_implemented"]
+
+    def read_headers(self, headers: Collection[str]) -> None:
+        """ Reads functions from source header files into a Dict[str, FunctionInfo] """
```

Comment:
There's no `Dict[str, FunctionInfo]` in the interface of this method. ITYM something like
```suggestion
        """ Reads functions to be wrapped from source header files."""
```

Reply:
I was trying to be more helpfull to people reviewing the code. 

read_function_declarations is written as c-code with input and output parameter rather than return.

    def read_function_declarations(functions: Dict[str, FunctionInfo],
                                   filename: str) -> None:

People would expect it to be like:

    def read_function_declarations(filename: str) -> [None](functions: Dict[str, FunctionInfo]):

So this comment was meant to explain it.  I could word it like that? 

> """Reads functions to be wrapped from source header files into self.functions."""

Reply:
This comment is on `read_headers`, not on `read_function_declarations` (which hasn't changed).